lua/entities/terminator_nextbot_zambieglasselite/shared.lua

```diff
@@ -73,32 +92,84 @@ function ENT:Think()
 
 end
 
+ENT.iShards = 25
+
+local sound_Add = sound.Add
+sound_Add {
+	name = "nextbotZambies_GlassBreakEliteA",
+	level = 95,
+	pitch = 90,
+	sound = {
+		"physics/glass/glass_largesheet_break1.wav",
+		"physics/glass/glass_largesheet_break2.wav",
+		"physics/glass/glass_largesheet_break3.wav"
+	}
+}
+sound_Add {
+	name = "nextbotZambies_GlassBreakEliteB",
+	level = 90,
+	sound = {
+		"physics/glass/glass_sheet_break1.wav",
+		"physics/glass/glass_sheet_break2.wav",
+		"physics/glass/glass_sheet_break3.wav"
+	}
+}
+
+local SHARD_MODELS = {
+	"models/gibs/glass_shard01.mdl",
+	"models/gibs/glass_shard02.mdl",
+	"models/gibs/glass_shard03.mdl",
+	"models/gibs/glass_shard04.mdl",
+	"models/gibs/glass_shard05.mdl",
+	"models/gibs/glass_shard06.mdl",
+}
+local SHARD_MODELS_LENGTH = 6
+
+local string_lower = string.lower
+local tonumber = tonumber
+
+function ENT:KeyValue( sKey, sValue )
+	if string.lower( sKey ) == "ishards" then
+		local f = tonumber( sValue )
+		if f then self.iShards = f end
+		return
+	end
+	return BaseClass.KeyValue( self, sKey, sValue )
+end
+
+local EffectData = EffectData
+local util = util
+local util_Effect = util.Effect
+local ents_Create = ents.Create
+local math = math
+local math_random = math.random
+local VectorRand = VectorRand
+local AngleRand = AngleRand
+local math_Rand = math.Rand
+local util_SpriteTrail = util.SpriteTrail
+local SafeRemoveEntityDelayed = SafeRemoveEntityDelayed
+
 function ENT:GlassZambDie()
     local pos = self:WorldSpaceCenter()
 
-    self:EmitSound( "physics/glass/glass_largesheet_break" .. math.random( 1, 3 ) .. ".wav", 95, 90 )
-    self:EmitSound( "physics/glass/glass_sheet_break" .. math.random( 1, 3 ) .. ".wav", 90, 100 )
+    self:EmitSound "nextbotZambies_GlassBreakA"
+    self:EmitSound "nextbotZambies_GlassBreakB"
 
     local effectdata = EffectData()
     effectdata:SetOrigin( pos )
     effectdata:SetScale( 2 )
-    util.Effect( "GlassImpact", effectdata )
-
-    local glassGibs = {
-        "models/gibs/glass_shard01.mdl",
-        "models/gibs/glass_shard02.mdl",
-        "models/gibs/glass_shard03.mdl",
-        "models/gibs/glass_shard04.mdl",
-        "models/gibs/glass_shard05.mdl",
-        "models/gibs/glass_shard06.mdl",
-    }
-
-    for _ = 1, 25 do
-        local gib = ents.Create( "prop_physics" )
+    util_Effect( "GlassImpact", effectdata )
+
+	local iShards = self.iShards
+	if iShards <= 0 then return end
+	local flAngularVelocity = iShards * 40
+	local flVelocityMin, flVelocityMax = iShards * 32, iShards * 48
+    for _ = 1, iShards do
+        local gib = ents_Create "prop_physics"
         if IsValid( gib ) then
             pos = pos + VectorRand() * 15
-            pos.z = pos.z + math.random( -5, 5 )
-            gib:SetModel( table.Random( glassGibs ) )
+            pos.z = pos.z + math_random( -5, 5 )
+            gib:SetModel( SHARD_MODELS[ math_random( 1, SHARD_MODELS_LENGTH ) ] )

```

Comment:
```suggestion
            gib:SetModel( SHARD_MODELS[ math_random( 1, #SHARD_MODELS ) ] )

```
👍 to replacing table.Random